rsconnect/admin_actions.py

```diff
@@ -0,0 +1,338 @@
+            click.secho("\tUse `rsconnect-admin rebuild run --aborted` to retry the aborted rebuilds.")
+            click.secho("Aborted rebuilds:")
+            for guid in aborted_rebuilds:
+                click.secho("\t%s" % guid)
+                content_rebuild_store.set_content_item_rebuild_status(connect_server, guid, RebuildStatus.ABORTED)
+        return False
+
+    click.secho()
+    click.secho("%d/%d content rebuilds completed in %s" % (len(complete) + len(error), len(content_items), rounded_duration))
+    click.secho("Success = %d, Error = %d" % (len(complete), len(error)))
+    if len(error) > 0:
+        click.secho()
+        click.secho("There were failures during your rebuild.")
+        click.secho("\tUse `rsconnect-admin rebuild ls --status=ERROR` to list the failed content.")
+        click.secho("\tUse `rsconnect-admin rebuild logs --guid` to check the rebuild logs of a specific content rebuild.")
+        # click.secho()
+        # click.secho("Failed content:")
+        # for c in error:
+        #     click.secho("\tName: %s, GUID: %s" % (c['name'], c['guid']))
+        return False
+    return True
+
+
+def _rebuild_content_item(connect_server, content, timeout=None):
+    # Pending futures will still try to execute when ThreadPoolExecutor.shutdown() is called
+    # so just exit immediately if the current rebuild has been aborted.
+    # ThreadPoolExecutor.shutdown(cancel_futures=) isnt available until py3.9
+    if content_rebuild_store.aborted():
+            return
+
+    guid = content['guid']
+    content_rebuild_store.ensure_logs_dir(connect_server, guid)
+    content_rebuild_store.set_content_item_rebuild_status(connect_server, guid, RebuildStatus.RUNNING)
+    task_result = api.do_start_content_rebuild(connect_server, guid, content.get('bundle_id'))
+    task_id = task_result.json_data['task_id']
+    log_file = content_rebuild_store.get_rebuild_log(connect_server, guid, task_id)
+    with open(log_file, 'w') as log:
+        # emit_task_log raises an exception if exit_code != 0
+        api.emit_task_log(connect_server, guid, task_id,
+            log_callback=lambda line: log.write("%s\n" % line), abort_func=content_rebuild_store.aborted)
+
+    if content_rebuild_store.aborted():
+        return
+
+    # grab the updated content metadata from connect and update our store
+    updated_content = api.do_content_get(connect_server, guid)
+    content_rebuild_store.update_content_item(connect_server, guid, updated_content)
+    content_rebuild_store.set_content_item_rebuild_status(connect_server, guid, RebuildStatus.COMPLETE)
+
+
+
+def emit_rebuild_log(connect_server, guid, format, task_id=None):
+    log_file = content_rebuild_store.get_rebuild_log(connect_server, guid, task_id)
+    if log_file:
+        with open(log_file, 'r') as f:
+            for line in f.readlines():
+                if format == "json":
+                    yield json.dumps({"message": line}) + "\n"
```

Comment:
I considered this but opted against it because _sometimes_ the output from the task logs includes a timestamp and it felt wrong to add another one on the client side.  And then that also begs the question: do we mutate the incoming log lines by prefixing them with the timestamp when they are written to disk?